models/taskRequests.js

```diff
@@ -73,6 +73,87 @@ const fetchTaskRequestById = async (taskRequestId) => {
   };
 };
 
+const createRequest = async (data, authenticatedUsername) => {
+  const queryFieldPath = data.requestType === TASK_REQUEST_TYPE.CREATION ? "externalIssueUrl" : "taskId";
+  const queryValue = data.requestType === TASK_REQUEST_TYPE.CREATION ? data.externalIssueUrl : data.taskId;
+  const statusQueryValue =
+    data.requestType === TASK_REQUEST_TYPE.CREATION
+      ? [TASK_REQUEST_STATUS.PENDING, TASK_REQUEST_STATUS.APPROVED]
+      : [TASK_REQUEST_STATUS.PENDING];
+  const taskRequestsSnapshot = await taskRequestsCollection
+    .where(queryFieldPath, "==", queryValue)
+    .where("status", "in", statusQueryValue)
+    .get();
```

Comment:
What taskRequestsCollection contain?

Reply:
Task requests 
eg: 
{
        "createdAt": 1697634769878,
        "requestType": "CREATION",
        "createdBy": "ajeyak",
        "lastModifiedBy": "ajeyak",
        "taskTitle": "Every website should have a staging link ",
        "externalIssueUrl": "https://api.github.com/repos/Real-Dev-Squad/todo-action-items/issues/12",
        "lastModifiedAt": 1697634839227,
        "requestors": [
            "eChYAP0kUwLo4wQ1gqMV",
            "6bQp8LeEETS1y8gPdzxA"
        ],
        "users": [
            {
                "proposedStartDate": 1697378947055,
                "proposedDeadline": 1697983747055,
                "description": "description",
                "userId": "eChYAP0kUwLo4wQ1gqMV",
                "status": "APPROVED"
            },
            {
                "proposedStartDate": 1697378947055,
                "proposedDeadline": 1697983747055,
                "description": "description",
                "userId": "6bQp8LeEETS1y8gPdzxA",
                "status": "PENDING"
            }
        ],
        "status": "APPROVED",
        "id": "whvkNIIRFW5gPrYlaqe9"
    }